crates/api_common/src/request.rs

```diff
@@ -140,6 +145,59 @@ pub(crate) async fn fetch_pictrs(
   }
 }
 
+/// Purges an image from pictrs
+/// Note: This should often be coerced from a Result to .ok() in order to fail softly, because:
+/// - It might fail due to image being not local
+/// - It might not be an image
+/// - Pictrs might not be set up
+pub async fn purge_image_from_pictrs(
+  client: &ClientWithMiddleware,
+  settings: &Settings,
+  image_url: &Url,
+) -> Result<(), LemmyError> {
+  if let Some(pictrs_config) = settings.pictrs_config.to_owned() {
+    is_image_content_type(client, image_url).await?;
+
+    // Check to make sure hosts match
+    if settings
+      .get_hostname_without_port()?
+      .ne(image_url.host_str().unwrap_or(""))
+    {
+      return Err(LemmyError::from_message(&format!(
+        "Could not purge Pictrs image with incorrect domain: {}",
+        image_url.as_str()
+      )));
+    }
```

Comment:
Is this check necessary? If something is wrong, pictrs will return an error on its own. But if someone changes the instance url, this would prevent image purges for no reason.

Reply:
You're probably right, it's not necessary.